scripts/install-npm.js

```diff
@@ -2,15 +2,31 @@ const fs = require('fs');
 const path = require('path');
 
 const sourceDir = 'node_modules';
-const destDir = 'kbase-extension/static/ext_modules';
+const extToolsDir = 'kbase-extension/static/ext_modules';
+const extComponentsDir = 'kbase-extension/static/ext_components';
 
-const filesToCopy = [
+const toolsToCopy = [
     ['msw/lib/umd/index.js', 'msw/index.js'],
     ['msw/lib/umd/mockServiceWorker.js', 'msw/mockServiceWorker.js'],
 ];
 
-for (const [from, to] of filesToCopy) {
-    const destPath = `${destDir}/${to}`;
+const pkg = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
+
+const extraComponents = [
+    'datatables.net',
+    'datatables.net-buttons',
+    'requirejs',
+    'requirejs-domready'
+];
+
+const componentsToCopy = Object.keys(pkg.dependencies).concat(extraComponents);
+
+for (const [from, to] of toolsToCopy) {
+    const destPath = `${extToolsDir}/${to}`;
     fs.mkdirSync(path.dirname(destPath), { recursive: true });
     fs.copyFileSync(`${sourceDir}/${from}`, destPath);
 }
+
+for (const component of componentsToCopy) {
+    fs.cpSync(`${sourceDir}/${component}`, `${extComponentsDir}/${component}`, { recursive: true });
```

Comment:
the `cp` function is in "Experimental" stability according to the fs docs (might change and be backward incompatible, https://nodejs.org/api/fs.html#fspromisescpsrc-dest-options) , but seems to just be a passthrough to the usual `cp` command in the file system.